### PR TITLE
fix error image use when using --image-mirror-country flag

### PR DESCRIPTION
**Pull request by scaat**
minikube version: v1.18.1 ~ v1.22.0

for example:

The `storage-provisioner` default image is `gcr.io/k8s-minikube/storage-provisioner`, registry is `gcr.io` and image is `k8s-minikube/storage-provisioner`. 

When I use the `--image-mirror-country=cn` flag to start a cluster, the image will be change to `registry.cn-hangzhou.aliyuncs.com/google_containers/k8s-minikube/storage-provisioner`, registry is `registry.cn-hangzhou.aliyuncs.com/google_containers`, image still is `k8s-minikube/storage-provisioner`.

The key to the problem is that the default registry and the registry using --image-mirror-country=cn are not one-to-one, so when using `--image-mirror-country=cn`, the image name needs to be processed.

In this case, should change `k8s-minikube/storage-provisioner` to `storage-provisioner` when using `--image-mirror-country=cn`

### CODE REVIEW

File: pkg/minikube/assets/addons.go [hunk 1]
```diff
@@ -724,9 +724,17 @@ func GenerateTemplateData(addon *Addon, cfg config.KubernetesConfig) interface{}
 				"image": strings.Split(image, "@")[0],
 			})
 		} else if opts.ImageRepository != "" {
+			tmp := strings.Split(image, "/")
+			var realImage string
```
Comment (medyagh) [inline pkg/minikube/assets/addons.go:728]: for code readability do u mind commenting on this what it means RealImage ? 
preferably with a link to a github issue that why we do this ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/assets/addons.go (hunk 1 above)
Comment (scaat) [reply pkg/minikube/assets/addons.go:728]: As I described above.

> When I use the --image-mirror-country=cn flag to start a cluster, the storage-provisioner pod will use the
registry.cn-hangzhou.aliyuncs.com/google_containers/k8s-minikube/storage-provisioner:v4 image, but the correct image should be
registry.cn-hangzhou.aliyuncs.com/google_containers/storage-provisioner:v4

the real image should `storage-provisioner:v4`, not `k8s-minikube/storage-provisioner:v4` when using `--image-mirror-country=cn` flag. @medyagh

Comment (medyagh) [reply pkg/minikube/assets/addons.go:728]: @scaat would u plz add the comment in the code ?
how about a unit test for this ?

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
@scaat are you still working on this PR ?

**Comment by medyagh** [comment]
@scaat are you still working on this PR ?

**Comment by scaat** [comment]
The latest version has solved this problem. thanks @medyagh

**Comment by scaat** [comment]
@medyagh yes

**Comment by scaat** [comment]
I made new modifications and instructions, please take a look.
Thanks. @medyagh

**Review by medyagh — CHANGES_REQUESTED** [review]
@scaat I think the problem you are describing should have been fixed with this PR
https://github.com/kubernetes/minikube/pull/11948

do u mind trying the binary on head to see if the problem is fixed?

**Comment by scaat** [comment]
@medyagh I have tryed the binary on head, the pr #11948 doesn't fix the problem.

**Comment by medyagh** [comment]
@scaat thank you for this PR, do u mind adding unit tests for this ? this seems like a great thing to add unit test for

**Comment by medyagh** [comment]
@scaat please also fix the `make lint` error

**Comment by scaat** [comment]
@medyagh Unit tests have been added and lint has been fixed. thanks

**Comment by scaat** [comment]
Can anyone help to review?

**Comment by medyagh** [comment]
@scaat sorry for the late responses in the PR review, we merged a couple PRs on china mirror and image mirros?

do u mind pulling upstream minikube and check if the problem your PR was trying to solve is still there on minikube HEAD and do we still need this PR ?

**Pull request closed without merging**